receiver/dockerstatsreceiver/metadata.yaml

```diff
@@ -220,14 +220,6 @@ metrics:
       value_type: int
       aggregation: cumulative
       monotonic: true
-  container.memory.swap:
```

Comment:
Ideally, such changes should be done via a deprecation process. 

1. Add a deprecation message in the metadata.yaml to generate the deprecation warnings:
```yaml
    warnings:
      if_enabled: This metric is deprecated and will be removed after v0.80.0
```

Reply:
I was thinking about this too, but if the metrics were never reported and since the component is alpha, it would be ok to do this in one step.

Reply:
We're ok with both approaches but as mentioned these metrics were never reported in any of the API versions AFAIK.
This is more of a cleanup task to avoid confusion.
But again, we're ok either way.

Reply:
These metrics were disabled by default. Someone could've added them to their configuration with:

```yaml
receivers:
  docker_stats:
    metrics:
      container.memory.swap:
        enabled: true
      container.memory.total_swap:
        enabled: true
```

and the user might never even realize that these metrics are not actually reported. When they upgrade, their configuration will be broken. To fix this, they would need to remove those `metrics` entries from the configuration, which doesn't sound like a lot to ask for. But on the other hand, perhaps we should be super nice and issue a warning before removing these metrics? I'm on a fence with this :smile:

Reply:
For such a niche situation I think we can lean on the Alpha stability here an do a breaking change in 1 go.  @carlossscastro can you share the collector startup error message that appears if you try to start the collector with these metrics enabled?

Reply:
@TylerHelmuth 

There is no error on the collector when they are enabled but they are never reported.
For memory metrics the receiver iterates through a map of metric names/values returned by the docker API and matches them with a map of names/recorders that we want to report. Since those 2 metric names are not returned by the api the recorder method for them was never called which avoided any errors:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/receiver.go#L160-L199